hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/expression/misc/CastFunctionIntegrationTest.java

```diff
@@ -16,6 +16,7 @@
 
 package com.hazelcast.jet.sql.impl.expression.misc;
 
+import com.hazelcast.core.HazelcastJsonValue;
```

Comment:
This class should test all types in combination with JSON on both sides (as source and target) and assert that it fails and what the error says.